client/src/Components/Layout/Login/Login.jsx

```diff
@@ -0,0 +1,331 @@
+import React, { Component } from 'react';
+import queryString from 'query-string';
+import { Link, Redirect } from 'react-router-dom';
+import {
+  Grid,
+  Box,
+  Typography,
+  Button,
+  Paper,
+  TextField,
+  FormHelperText,
+  FormControl,
+  Input,
+  InputLabel,
+  IconButton,
+  InputAdornment,
+} from '@material-ui/core';
+import {
+  Person as PersonIcon,
+  Lock as LockIcon,
+  Visibility,
+  VisibilityOff,
+} from '@material-ui/icons';
+import Cookies from 'js-cookie';
+import { withStyles } from '@material-ui/core/styles';
+
+import LoaderProgress from '../../Common/LoaderProgress';
+
+import axios from 'axios';
+import LoginStyles from './LoginStyle';
+
+class Login extends Component {
+  state = {
+    logindata: { email: null, password: null },
+    isLoading: false,
+    displayBlock: true,
+    showPassword: false,
+    email: {
+      Message: '',
+      isValid: true,
+    },
+    password: {
+      Message: '',
+      isValid: true,
+      minLength: 3,
+    },
+    redirect: false,
+    ReturnUrlText: '/',
+    SignupLink: '/user/Signup/',
+  };
+
+  componentDidMount() {
+    const AuthToken = Cookies.get('AuthToken');
+
+    console.log(AuthToken);
+    console.log(!AuthToken);
+    const qstring = queryString.parse(this.props.history.location.search);
+    if (qstring.ReturnUrl) {
+      this.setState({
+        ReturnUrlText: qstring.ReturnUrl,
+        SignupLink: `/user/Signup/?ReturnUrl=${qstring.ReturnUrl}`,
```

Comment:
This is a nice touch - it doesn't matter too much but you probably don't need to store this with setState - this will make React run it's reconciliation process, but if you just store it as a property of the class that you can access when you need it won't try to re-render

Reply:
i save in state to use in render Action , i need to return the user from where he comes , when the user Wants to enroll  in event , user must be login  , if not log in he must go to login and back to event details page to contuin and take place in event